corehq/apps/locations/views.py

```diff
@@ -407,12 +412,40 @@ def products_form(self):
 
         form = MultipleSelectionForm(
             initial={'selected_ids': self.products_at_location},
-            submit_label=_("Update Product List")
+            submit_label=_("Update Product List"),
+            prefix="products",
         )
         form.fields['selected_ids'].choices = self.all_products
         return form
 
     @property
+    @memoized
+    def all_users(self):
```

Comment:
My opinion is that it's worth it to switch this one and it doesn't bother me too much to have them backed by separate things (and probably worth the performance hit to keep the location filter by elastic unless we're already going to load the users every time anyways). 

I don't feel strongly enough about it to block merge though.

I think my more general position is we shouldn't build new features on elasticsearch unless either:

1. it's obviously better (as the location query would be)
2. we gain increased confidence that our indices and process are not dropping data or significantly delayed

In this case unless you ran performance numbers indicating that elastic was much faster than couch, just using the tried-and-true things seems better. It also isolates the code/query to a single well-tested thing that is consistent that everyone uses which is nice for other reasons.